climada/hazard/tc_tracks_forecast.py

```diff
@@ -179,76 +184,156 @@ def fetch_bufr_ftp(target_dir=None, remote_dir=None):
 
         return localfiles
 
-    def read_one_bufr_tc(self, file, id_no=None, fcast_rep=None):
+    def read_one_bufr_tc(self, file, id_no=None):
         """ Read a single BUFR TC track file.
 
         Parameters:
             file (str, filelike): Path object, string, or file-like object
             id_no (int): Numerical ID; optional. Else use date + random int.
-            fcast_rep (int): Of the form 1xx000, indicating the delayed
-                replicator containing the forecast values; optional.
-        """
-
-        decoder = pybufrkit.decoder.Decoder()
-
-        if hasattr(file, 'read'):
-            bufr = decoder.process(file.read())
-        elif hasattr(file, 'read_bytes'):
-            bufr = decoder.process(file.read_bytes())
-        elif Path(file).is_file():
-            with Path(file).open('rb') as i:
-                bufr = decoder.process(i.read())
+        """     
+        # Open the bufr file
+        bufr = ec.codes_bufr_new_from_file(file)
+        # we need to instruct ecCodes to expand all the descriptors
+        # i.e. unpack the data values
+        ec.codes_set(bufr, 'unpack', 1)
+        
+        # get the forcast time
+        timestamp_origin = dt.datetime(
+            ec.codes_get(bufr, 'year'), ec.codes_get(bufr, 'month'),
+            ec.codes_get(bufr, 'day'), ec.codes_get(bufr, 'hour'),
+            ec.codes_get(bufr, 'minute'),
+        )
+        timestamp_origin = np.datetime64(timestamp_origin)
+        
+        # number of timestep
+        n_timestep = 0
+        while True:
+            n_timestep = n_timestep + 1
```

Comment:
Could you read the number of times steps from the file first? That would make a cleaner loop.

Reply:
alternately, could you run the loop starting at line 250 without knowing n_timesteps first?
i.e. replace the for loop at line 250 with this while loop?

Reply:
I am a bit worried about the while loop which does not look like it has a very clear stopping point.